geode-cq/src/distributedTest/java/org/apache/geode/cache/query/dunit/QueryMonitorDUnitTest.java

```diff
@@ -358,6 +363,48 @@ public void testCacheOpAfterQueryCancel() throws Exception {
     });
   }
 
+  @Test
+  public void testQueryObjectReusableAfterFirstExecutionTimesOut() throws Exception {
+    server1.invoke(() -> {
+      // Setting query timeout to 10 seconds
+      GemFireCacheImpl.MAX_QUERY_EXECUTION_TIME = 10000;
```

Comment:
This test will take 10s to run. That seems long to me.

Could you set this value to say `100` (100ms) until the first query times out, and then set it to `10_000` (10s) before issuing the second query (the one meant to succeed)?

Reply:
This is a bummer, but MAX_QUERY_EXECUTION_TIME cannot be changed once the singleton query monitor is constructed.  A lame implementation detail that I think is remediable, but my initial thought was this PR probably isn't the place to change the query monitor construction behavior.  I'd actually like to do other things like DI the query monitor which would go into that same PR which fixes the unchangeable MAX_QUERY_EXECUTION_TIME issue.  I agree 10 seconds is a bit much, but I erred on the side of not wanting flakey failures due to the query actually taking 10 seconds in the successful case.  And DUnit tests take forever with initialization and cleanup anyways (not that thats a very good excuse, just sayin).